src/main/java/org/broadinstitute/hellbender/utils/downsampling/PositionalDownsampler.java

```diff
@@ -71,16 +73,31 @@ private void handlePositionalChange( final GATKRead newRead ) {
         // Use ReadCoordinateComparator to determine whether we've moved to a new start position.
         // ReadCoordinateComparator will correctly distinguish between purely unmapped reads and unmapped reads that
         // are assigned a nominal position.
-        if ( previousRead != null && ReadCoordinateComparator.compareCoordinates(previousRead, newRead, header) != 0 ) {
-            if ( reservoir.hasFinalizedItems() ) {
-                finalizeReservoir();
+        if ( previousRead != null) {
+            final int cmpDiff = ReadCoordinateComparator.compareCoordinates(previousRead, newRead, header);
+            if (cmpDiff == 1) {
+                throw new GATKException.ShouldNeverReachHereException(
```

Comment:
Right. BTW, I originally added this check because the `HaplotypeCallerSpark` tests were getting into this state after I made my changes, but before I added the code that resets `previousRead` to null in `finalizeReservoir` (see my response to your comment about that below).